src/AppBundle/Entity/SurveyNotification.php

```diff
@@ -0,0 +1,178 @@
+<?php
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+
+/**
+ * @ORM\Entity(repositoryClass="AppBundle\Entity\Repository\SurveyNotificationRepository")
+ * @ORM\Table(name="survey_notification")
+ * @UniqueEntity(
+ *      fields={"userIdentifier"}
+ * )
+ */
+class SurveyNotification
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var User
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\User")
+     */
+    private $user;
+
+    /**
+     * @var SurveyLinkClick[]
+     * @ORM\OneToMany(targetEntity="SurveyLinkClick", mappedBy="notification")
+     */
+    private $surveyLinkClicks;
+
+    /**
+     * @var SurveyNotifier
+     * @ORM\ManyToOne(targetEntity="SurveyNotifier", inversedBy="surveyNotification")
+     */
+    private $surveyNotifier;
+
+
+    /**
+     * @var \DateTime
+     * @ORM\Column(name="time_notification_Sent", type="datetime", nullable=true)
+     */
+    private $timeNotificationSent;
+
+
+    /**
+     * @var string
+     * @ORM\Column(name="user_identifier", type="string", unique=true)
+     */
+    private $userIdentifier;
+
+    /**
+     * @var bool
+     * @ORM\Column(type="boolean")
+     */
+    private $sent;
+
+
+    public function __construct()
+    {
+        $this->userIdentifier = bin2hex(openssl_random_pseudo_bytes(3));
```

Comment:
Hvor mange tegn vil bli generert her? Hvis dette brukes til å genrere en link som kan besøkes uten å kreve innlogging, hvor sikker er denne mot brute force?

Reply:
Enig at det blir litt for lett mtp. bruteforce her, bytter til 
```suggestion
        $this->userIdentifier = bin2hex(openssl_random_pseudo_bytes(12));
```
som det brukes i email-subscriber